src/gui/widgets/TrackLabelButton.cpp

```diff
@@ -64,8 +64,17 @@ TrackLabelButton::TrackLabelButton( TrackView * _tv, QWidget * _parent ) :
 	}
 
 	setIconSize( QSize( 24, 24 ) );
-	setText( " " );
+	if( ConfigManager::inst()->value( "ui", "compacttrackbuttons" ).toInt() )
```

Comment:
Okay, we change the elide width dynamically to the button width. And call the nameChanged slot on resize. 

If the buttons width smaller the a certain value we set the text to "".